core/src/main/java/org/elasticsearch/monitor/fs/FsProbe.java

```diff
@@ -42,12 +54,77 @@ public FsInfo stats() throws IOException {
         NodePath[] dataLocations = nodeEnv.nodePaths();
         FsInfo.Path[] paths = new FsInfo.Path[dataLocations.length];
         for (int i = 0; i < dataLocations.length; i++) {
-            paths[i] = getFSInfo(dataLocations[i]);
+            paths[i] = getQuotaAwareFSInfo(dataLocations[i]);
         }
         return new FsInfo(System.currentTimeMillis(), paths);
     }
 
-    public static FsInfo.Path getFSInfo(NodePath nodePath) throws IOException {
+    public FsInfo.Path getQuotaAwareFSInfo(NodePath nodePath) throws IOException {
+        Optional<Path> closestMatchingQuota = quotas.keySet().stream().filter((path) -> nodePath.path.startsWith(path))
```

Comment:
The entire filestore only reports data for the entire volume. The whole point here is to report according to a quota specific to a path.